From 9f4aa6ebf2193bb19dd8012de7699fce35681078 Mon Sep 17 00:00:00 2001
From: Pascal Nowack <Pascal.Nowack@gmx.de>
Date: Mon, 6 Aug 2018 14:21:02 +0200
Subject: [PATCH] build: Fix build with meson >= 47.0

With meson >= 47.0, building gnome-shell will fail
when --rebuild-types is used as an option for gtk-doc,
so stop using that option and bump the meson requirement.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/191
---
 docs/reference/shell/meson.build | 3 +--
 docs/reference/st/meson.build    | 3 +--
 meson.build                      | 2 +-
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/docs/reference/shell/meson.build b/docs/reference/shell/meson.build
index 1469ed422..a32d6a58a 100644
--- a/docs/reference/shell/meson.build
+++ b/docs/reference/shell/meson.build
@@ -50,8 +50,7 @@ gnome.gtkdoc('shell',
     join_paths(meson.build_root(), 'src')
   ],
   scan_args: [
-    '--ignore-headers=' + ' '.join(private_headers + exclude_directories),
-    '--rebuild-types'
+    '--ignore-headers=' + ' '.join(private_headers + exclude_directories)
   ],
   install: true
 )
diff --git a/docs/reference/st/meson.build b/docs/reference/st/meson.build
index fe89b53d4..3e9e9ab7c 100644
--- a/docs/reference/st/meson.build
+++ b/docs/reference/st/meson.build
@@ -17,8 +17,7 @@ gnome.gtkdoc('st',
   ],
   scan_args: [
     '--ignore-headers=' + ' '.join(private_headers),
-    '--rebuild-sections',
-    '--rebuild-types'
+    '--rebuild-sections'
   ],
   install: true
 )
diff --git a/meson.build b/meson.build
index 03b9d52a8..e33a76183 100644
--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,6 @@
 project('gnome-shell', 'c',
   version: '3.28.3',
-  meson_version: '>= 0.42.0',
+  meson_version: '>= 0.47.0',
   license: 'GPLv2+'
 )
 
-- 
2.17.0

